Unable to disable annotation rule for annotations with arguments

### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Describe the bug

Given the following in the config
```swift
attributes:
  always_on_same_line: ["@IBAction", "@NSManaged", "@objc", "@objcMembers"]
```

The code below has a warning stating it should be on its own line
```swift
@objc(EMGridMetrics) class GridMetrics: NSObject {
  // …
}
```

```bash
$ swiftlint lint

…/GridMetrics.swift:1:22: warning: Attributes Violation: Attributes should be on their own lines in functions and types, but on the same line as variables and imports. (attributes)
```

### Environment

* SwiftLint version (run `swiftlint version` to be sure)? **0.40.0**
* Installation method used (Homebrew, CocoaPods, building from source, etc)? **Homebrew**
* Paste your configuration file:

```yml
included:
  - Sources
opt_in_rules:
  - attributes

attributes:
  always_on_same_line: ["@IBAction", "@NSManaged", "@objc", "@objcMembers"]
```

* Are you using [nested configurations](https://github.com/realm/SwiftLint#nested-configurations)? No
* Which Xcode version are you using (check `xcodebuild -version`)? Xcode 11, and Xcode 12
* Do you have a sample that shows the issue? See above

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -69,6 +69,10 @@
   [JP Simard](https://github.com/jpsim)
   [#3306](https://github.com/realm/SwiftLint/issues/3306)
 
+* Fix false positives with parameterized attributes in `attributes`.  
+  [JP Simard](https://github.com/jpsim)
+  [#3316](https://github.com/realm/SwiftLint/issues/3316)
+
 ## 0.40.3: Greased Up Drum Bearings
 
 #### Breaking

--- a/Source/SwiftLintFramework/Rules/Style/AttributesRule.swift
+++ b/Source/SwiftLintFramework/Rules/Style/AttributesRule.swift
@@ -96,7 +96,13 @@ public struct AttributesRule: ASTRule, OptInRule, ConfigurationProviderRule {
         let tokens = file.syntaxMap.tokens(inByteRange: line.byteRange)
         let attributesTokensWithRanges = tokens.compactMap { attributeName(token: $0, file: file) }
 
-        let attributesTokens = Set(attributesTokensWithRanges.map { $0.0 })
+        let attributesTokens = Set(
+            attributesTokensWithRanges.map { tokenString, _ in
+                // Some attributes are parameterized, such as `@objc(name)`, so discard anything from an opening
+                // parenthesis onward.
+                String(tokenString.prefix(while: { $0 != "(" }))
+            }
+        )
 
         do {
             let previousAttributesWithParameters = try attributesFromPreviousLines(lineNumber: lineNumber - 1,

--- a/Source/SwiftLintFramework/Rules/Style/AttributesRuleExamples.swift
+++ b/Source/SwiftLintFramework/Rules/Style/AttributesRuleExamples.swift
@@ -85,7 +85,6 @@ internal struct AttributesRuleExamples {
         Example("@IBDesignable ↓class MyCustomView: UIView {}"),
         Example("@testable\n↓import SourceKittenFramework"),
         Example("@testable\n\n\n↓import SourceKittenFramework"),
-        Example("@objc(foo_x) ↓var x: String"),
         Example("@available(iOS 9.0, *) @objc(abc_stackView)\n ↓let stackView: UIStackView"),
         Example("@objc(abc_addSomeObject:) @NSManaged\n ↓func addSomeObject(book: SomeObject)"),
         Example("@objc(abc_addSomeObject:)\n @NSManaged\n ↓func addSomeObject(book: SomeObject)"),

--- a/Tests/SwiftLintFrameworkTests/AttributesRuleTests.swift
+++ b/Tests/SwiftLintFrameworkTests/AttributesRuleTests.swift
@@ -18,6 +18,9 @@ class AttributesRuleTests: XCTestCase {
                 @objc private var object: RLMWeakObjectHandle?
                 @objc private var property: RLMProperty?
             }
+            """),
+            Example("""
+            @objc(XYZFoo) class Foo: NSObject {}
             """)
         ]
         let triggeringExamples = [

